sql/src/main/java/io/crate/cluster/gracefulstop/DecommissioningService.java

```
@@ -201,54 +202,29 @@ public void clusterChanged(ClusterChangedEvent event) {
             DECOMMISSION_PREFIX + clusterService.localNode().getId(), true).build();
-        updateSettingsAction.execute(new ClusterUpdateSettingsRequest().transientSettings(settings), new ActionListener<ClusterUpdateSettingsResponse>() {
-            @Override
-            public void onResponse(ClusterUpdateSettingsResponse clusterUpdateSettingsResponse) {
-                // changing settings triggers AllocationService.reroute -> shards will be relocated
-
-                // NOTE: it waits for ALL relocating shards, not just those that involve THIS node.
-                ClusterHealthRequest request = new ClusterHealthRequest()
-                    .waitForNoRelocatingShards(true)
-                    .waitForEvents(Priority.LANGUID)
-                    .timeout(gracefulStopTimeout);
-
-                if (dataAvailability == DataAvailability.FULL) {
-                    request = request.waitForGreenStatus();
-                } else {
-                    request = request.waitForYellowStatus();
-                }
 
-                final long startTime = System.nanoTime();
-
-                healthAction.execute(request, new ActionListener<ClusterHealthResponse>() {
-                    @Override
-                    public void onResponse(ClusterHealthResponse clusterHealthResponse) {
-                        executorService.submit(() -> exitIfNoActiveRequests(startTime));
-                    }
-
-                    @Override
-                    public void onFailure(Exception e) {
-                        executorService.submit(() -> forceStopOrAbort(e));
-                    }
-                });
-            }
-
-            @Override
-            public void onFailure(Exception e) {
+        return clusterUpdateSettings(settings)
+            .exceptionally(e -> {
                 logger.error("Couldn't set settings. Graceful shutdown failed", e);
-            }
-        });
+                return null;
```

I think the error needs to be re-raised, otherwise we swallow the error and continue in the chain